src/v/archival/ntp_archiver_service.cc

```diff
@@ -1900,4 +1901,11 @@ void ntp_archiver::complete_transfer_leadership() {
     _leader_cond.signal();
 }
 
+bool ntp_archiver::local_storage_pressure() const {
+    auto eviction_offset = _parent.eviction_requested_offset();
+    return eviction_offset.has_value()
+           && _parent.archival_meta_stm()->get_last_clean_at()
+                < eviction_offset.value();
```

Comment:
This condition should use `<=`. The requests for truncation that come into `log_eviction_stm` are inclusive (i.e. up until an including the specified offset). Raft truncates at the offset after the requested one (see [here](https://github.com/redpanda-data/redpanda/blob/be36dc66e876fbcfe0365b68517ddcefa245b0b5/src/v/raft/consensus.cc#L2217-L2218)).

Reply:
Great catch.  Pushed an update.